src/models/files/node-file-stream.js

```
@@ -3,6 +3,14 @@ const errors = require('../../errors');
 const fs = require('fs');
 const os = require('os');
 const path = require('path');
+const mkdirp = require('mkdirp');
+const rimraf = require('rimraf');
+const config = require('../../config');
+const TinyDb = require('../../db/tiny-db');
+const cryptoUtil = require('../../crypto/util');
+
+const DEFAULT_TEMP_ROOT = path.join(os.tmpdir(), config.appId);
```

Let's use `path.join(config.StorageEngine.storageFolder, "ImageCache")` (or something else instead of "ImageCache", but not "Cache" because it's taken by Electron) if that's okay for mobile.

---

Actually, getting `storageFolder` isn't a good idea, instead how about `FileStream` will have a configurable value `cacheFolder` and then desktop and mobile will set it in config just like they do, e.g. desktop:

```
cfg.FileStream = FileStream;
cfg.FileStream.cacheFolder = path.join(app.getPath('userData'), 'ImageCache');
cfg.StorageEngine = StorageEngine;
cfg.StorageEngine.storageFolder = app.getPath('userData');
```

---

and if it's not set, just throw an error instead of having a bad default.